storage/src/store/base.rs

```diff
@@ -0,0 +1,845 @@
+//! A simple append-only key-value [`Store`], based on a log of state-change operations backed by a [`Journal`].
+
+use crate::{
+    index::Index,
+    journal::fixed::{Config as JConfig, Journal},
+    store::{operation::Fixed, Error, UpdateResult},
+    translator::Translator,
+};
+use commonware_runtime::{buffer::PoolRef, Clock, Metrics, Storage as RStorage};
+use commonware_utils::Array;
+use futures::{pin_mut, StreamExt};
+use tracing::{debug, warn};
+
+/// The size of the read buffer to use for replaying the operations log when rebuilding the
+/// snapshot.
+const SNAPSHOT_READ_BUFFER_SIZE: usize = 1 << 16;
+
+/// Configuration for initializing a [`Store`] database.
+#[derive(Clone)]
+pub struct Config<T: Translator> {
+    /// The name of the [`RStorage`] partition used to persist the (pruned) log of operations.
+    pub log_journal_partition: String,
+
+    /// The items per blob configuration value used by the log journal.
+    pub log_items_per_blob: u64,
+
+    /// The size of the write buffer to use for each blob in the log journal.
+    pub log_write_buffer: usize,
+
+    /// The [`Translator`] used by the compressed index.
+    pub translator: T,
+
+    /// The buffer pool to use for caching data.
+    pub buffer_pool: PoolRef,
+}
+
+/// A simple append-only key-value store backed by a log of operations.
+///
+/// The [`Store`] provides persistent key-value storage with transactional semantics and automatic
```

Comment:
nit: Again, prefer describing `store` as a standalone thing rather than a lighter adb::any (and adb::any as a more powerful version of this, which is more primitive).